cinema/serializers.py

```diff
@@ -74,7 +121,55 @@
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_seats = serializers.SerializerMethodField()
+
+    @staticmethod
+    def get_taken_seats(obj):
+        return [{
+            "row": ticket.row,
+            "seat": ticket.seat
+        } for ticket in obj.tickets.all()]
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = (
+            "id",
+            "show_time",
+            "movie",
+            "cinema_hall",
+            "taken_seats",
+            "tickets_available"
+        )
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    movie_session = MovieSessionListSerializer(many=False, read_only=True)
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+        validators = [
+            UniqueTogetherValidator(
+                queryset=Ticket.objects.all(), fields=[
+                    "row",
+                    "seat",
+                    "movie_session"
+                ]
+            )
+        ]
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(many=True, read_only=False, allow_empty=False)
```

Comment:
The `tickets` field in `OrderSerializer` is set to `read_only=False` and `allow_empty=False`, which means it expects input data. Ensure that the input data for `tickets` is validated correctly before processing in the `create` method.